Sources/Datadog/FeaturesIntegration/WebView/WebRUMEventConsumer.swift

```diff
@@ -29,32 +29,87 @@ internal class WebRUMEventConsumer: WebEventConsumer {
+            let mappedViewEvent = mapIfNeeded_RUMViewEvent(dataModel: viewEvent, context: rumContext, offset: getOffset(viewID: viewEvent.view.id))
             write(mappedViewEvent)
         case "action":
             let actionEvent = try jsonDecoder.decode(RUMActionEvent.self, from: eventData)
-            let mappedActionEvent = mapIfNeeded(dataModel: actionEvent, context: rumContext, offset: getOffset(viewID: actionEvent.view.id))
+            let mappedActionEvent = mapIfNeeded_RUMActionEvent(dataModel: actionEvent, context: rumContext, offset: getOffset(viewID: actionEvent.view.id))
             write(mappedActionEvent)
         case "resource":
             let resourceEvent = try jsonDecoder.decode(RUMResourceEvent.self, from: eventData)
-            let mappedResourceEvent = mapIfNeeded(dataModel: resourceEvent, context: rumContext, offset: getOffset(viewID: resourceEvent.view.id))
+            let mappedResourceEvent = mapIfNeeded_RUMResourceEvent(dataModel: resourceEvent, context: rumContext, offset: getOffset(viewID: resourceEvent.view.id))
             write(mappedResourceEvent)
         case "error":
             let errorEvent = try jsonDecoder.decode(RUMErrorEvent.self, from: eventData)
-            let mappedErrorEvent = mapIfNeeded(dataModel: errorEvent, context: rumContext, offset: getOffset(viewID: errorEvent.view.id))
+            let mappedErrorEvent = mapIfNeeded_RUMErrorEvent(dataModel: errorEvent, context: rumContext, offset: getOffset(viewID: errorEvent.view.id))
             write(mappedErrorEvent)
         case "long_task":
             let longTaskEvent = try jsonDecoder.decode(RUMLongTaskEvent.self, from: eventData)
-            let mappedLongTaskEvent = mapIfNeeded(dataModel: longTaskEvent, context: rumContext, offset: getOffset(viewID: longTaskEvent.view.id))
+            let mappedLongTaskEvent = mapIfNeeded_RUMLongTaskEvent(dataModel: longTaskEvent, context: rumContext, offset: getOffset(viewID: longTaskEvent.view.id))
             write(mappedLongTaskEvent)
         default:
             userLogger.error("🔥 Web RUM Event Error - Unknown event type: \(eventType)")
         }
     }
 
-    private func mapIfNeeded<T: RUMDataModel>(dataModel: T, context: RUMContext?, offset: Offset) -> T {
-        // TODO: RUMM-1786 implement mutating session_id & application_id
-        return dataModel
+    private func mapIfNeeded_RUMViewEvent(dataModel: RUMViewEvent, context: RUMContext?, offset: Offset) -> RUMViewEvent {
+        guard let context = context else {
+            return dataModel
+        }
+        var mappedDataModel = dataModel
+        mappedDataModel.application.id = context.rumApplicationID
+        mappedDataModel.session.id = context.sessionID.toRUMDataFormat
+        mappedDataModel.date = mappedDataModel.date + getOffset(viewID: mappedDataModel.view.id)
+        mappedDataModel.dd.session?.plan = .plan1
+        return mappedDataModel
```

Comment:
Android also mutates `service` property as can be seen [here](https://github.com/DataDog/dd-sdk-android/blob/feature/hybrid-application/dd-sdk-android/src/main/kotlin/com/datadog/android/webview/internal/rum/WebViewRumEventMapper.kt#L28)

in iOS we send `service: nil` in `RUMViewScope`, so i don't think we should mutate it. wdyt @ncreated ?